proposals/0082-relax-transaction-constraints.md

```diff
@@ -0,0 +1,284 @@
+---
+simd: '0082'
+title: Relax Transaction Constraints
+authors:
+  - Andrew Fitzgerald (Solana Labs)
+category: Standard
+type: Core
+status: Draft
+created: 2023-10-30
+feature:
+---
+
+## Summary
+
+Transaction errors fall into two categories: protocol violating errors and
+runtime errors.
+Protocol violation errors are those that break some constraint of the protocol,
+and any blocks containing such transactions must be rejected by the network.
+Runtime errors are those that occur during the processing of a transaction, and
+generally these errors mean that state changes are limited to a few accounts,
+but the transaction itself may be included in a block.
+Any block containing a transaction that results in a runtime error is still
+valid.
+This proposal aims to change several protocol violation errors into runtime
+errors, in order to simplify the protocol, and give more flexibility to
+block-producer and block-validator implementations.
+
+## Motivation
+
+The current protocol places many constraints on the structure and contents
+of blocks; if any constraints are broken, block-validators will mark the
+entire block as invalid.
+Many of these constraints are necessary, but some of them are not, and lead to
+additional complexity in the protocol.
+This proposal aims to relax some of the constraints at the individual
+transaction level, in order to simplify the protocol, and give more flexibility
+to block-producer and block-validator implementations.
+
+More specifically, this proposal changes several protocol violation errors into
+runtime errors.
+This means that transactions that would previously be dropped with an error,
+can now be included in a block, but will either not be executed at all, or
+will have limited state changes.
+The goal is to remove much of the reliance on account-state in order to
+validate a block.
+This proposal on its' own, will not enable asynchronous execution, but it will
+remove one of the barriers to asynchronous execution.
+
+## Alternatives Considered
+
+1. Do nothing
+    - This is the simplest option, as we could leave the protocol as is.
+    However, this leaves the protocol more complex than it needs to be.
+2. Also relax fee-paying constraint
+    - This was considered, and included in the intially reviewed proposal.
+    However, this was decided to be moved to a separate follow-up proposal,
+    should this one be accepted.
+    There was significant disagreements on how exactly transactions that cannot
+    pay fees should be handled.
+    By keeping fee-paying as a constraint for the current proposal, it will
+    allow this proposal to be accepted more quickly, which will give
+    significant benefits to the network.
+3. Additionally, relax the address lookup table resolution constraint
+    - This was considered, since it is a transaction-level constraint that is
+    depdendent on account-state. However, due to entry-level and block-level
```

Comment:
small typo here @apfitzge 

"depdendent"

Reply:
Thanks! Found a few other spelling, consistency, or grammar issues while fixing it.
Pushed the fixes.